rotkehlchen/chain/bitcoin/__init__.py

```diff
@@ -9,62 +7,94 @@
     """
-    source = 'blockchain.info'
-    balances: Dict[BTCAddress, FVal] = {}
-    try:
-        if _have_bc1_accounts(accounts):
-            # if 1 account is bech32 we have to query blockstream. blockchaininfo won't work
-            source = 'blockstream'
-            balances = {}
-            for account in accounts:
-                url = f'https://blockstream.info/api/address/{account}'
-                response_data = request_get_dict(url=url, handle_429=True, backoff_in_seconds=4)
-                stats = response_data['chain_stats']
-                balance = int(stats['funded_txo_sum']) - int(stats['spent_txo_sum'])
-                balances[account] = satoshis_to_btc(balance)
-        else:
-            # split the list of accounts into sublists of 80 addresses per list to overcome:
-            # https://github.com/rotki/rotki/issues/3037
-            accounts_chunks = [accounts[x:x + 80] for x in range(0, len(accounts), 80)]
-            for accounts_chunk in accounts_chunks:
-                params = '|'.join(accounts_chunk)
-                btc_resp = request_get_dict(
-                    url=f'https://blockchain.info/multiaddr?active={params}',
-                    handle_429=True,
-                    # If we get a 429 then their docs suggest 10 seconds
-                    # https://blockchain.info/q
-                    backoff_in_seconds=10,
-                )
-                for entry in btc_resp['addresses']:
-                    balances[entry['address']] = satoshis_to_btc(FVal(entry['final_balance']))
-    except (
+    errors: dict[str, Exception] = {}
+    for name, api in _BITCOIN_BALANCE_APIS.items():
+        try:
+            balances = api(accounts)
+        except (
             requests.exceptions.RequestException,
             UnableToDecryptRemoteData,
             requests.exceptions.Timeout,
-    ) as e:
-        raise RemoteError(f'bitcoin external API request for balances failed due to {str(e)}') from e  # noqa: E501
-    except KeyError as e:
-        raise RemoteError(
-            f'Malformed response when querying bitcoin blockchain via {source}.'
-            f'Did not find key {e}',
-        ) from e
+            BchNotSupported,
+            RemoteError,
+        ) as e:
+            errors[name] = e
+            continue
+        except KeyError as e:
+            raise RemoteError(
+                f'Malformed response when querying bitcoin blockchain via {name}.'
+                f'Did not find key {e}',
+            ) from e
+        else:
+            return balances
 
-    return balances
+    serialized_errors = ', '.join(f'{source} failed with error: {str(error)}' for (source, error) in errors.items())  # noqa: E501
+    raise RemoteError(f'Bitcoin external API request for balances failed due to {serialized_errors}')  # noqa: E501
 
 
 def _check_blockstream_for_transactions(
-        accounts: List[BTCAddress],
-) -> Dict[BTCAddress, Tuple[bool, FVal]]:
```

Comment:
Why? We use typing everywhere

Reply:
It's not necessary to do that since python 3.9, it's the same if we do `List` and `list` in typehints, shouldn't matter as long as all of us are using Python 3.9 or above.

Reply:
What matters is consistency. Please be consistent and use whatever is used in the rest of the codebase